### PR TITLE
1.11 algorithms summary tasks

### PR DESCRIPTION
**Pull request by AlexandraPopovaz**

### CODE REVIEW

File: src/algorithms-task3.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    let lowerStr = "";
+    let amountLowerChars = 0;
+    let strLowerCase = str.toLowerCase();
+
+    for (let i = 0; i < strLowerCase.length; i++) {
+        let strBefore = str[i];
+        let strAfter = strLowerCase[i];
+
+        if (strBefore === strAfter && strBefore !== " ") {
+            lowerStr += strLowerCase[i];
```
Comment (viraxslot) [inline src/algorithms-task3.ts:58]: No need to add chars to the new string, you can just increase the resulting amount.

File: src/algorithms-task3.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    const arrayElements = Object.keys(myObj2);
+    arrayElements.forEach((el) => {
+        arrResult[myObj2[el].pos] = el;
+    });
+    return arrResult.join("");
+}
+
+/*
+Задача 3.
```
Comment (viraxslot) [inline src/algorithms-task3.ts:41]: English only please

File: src/algorithms-task2.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "2": 1,
+  "a": 3,
+  "b": 2,
+  "c": 1,
+  "d: 1
+}
+*/
+
+let stringExample2: any = "araadssgs";
+function example(stringExample2: string): Record<string, number> {
```
Comment (viraxslot) [inline src/algorithms-task2.ts:16]: stringExample2 - I'd rename the function parameter to "param" or something like this.

File: src/algorithms-task2.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    let result: Record<string, number> = {};
+    for (let i = 0; i < stringExample2.length; i++) {
+        if (result.hasOwnProperty(stringExample2[i])) {
```
Comment (viraxslot) [inline src/algorithms-task2.ts:19]: You can use `hasOwn` instead, it's the same but shorter.

File: src/algorithms-task3.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
+/*Write a function that will return a string for the presented object.
+
+Task 1.
+Input data: { "a": 2, "b": 1, "c": 3}
+Output data: aabccc
+*/
+
+let myObj = { j: 5, E: 5, f: 5, r: 5 };
+function getString(myObj: Record<string, number>): string | null {
```
Comment (viraxslot) [inline src/algorithms-task3.ts:9]: Same, I'd name the parameter in a neutral way: param.

File: src/algorithms-task3.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    let result: null | string = "";
+    Object.keys(myObj).forEach((el) => {
+        if (myObj[el] < 0) {
+            result = null;
+            return;
+        }
+        result += el;
```
Comment (viraxslot) [inline src/algorithms-task3.ts:16]: I don't understand why you're adding an element to the result here.

File: src/algorithms-task3.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        if (str[i] === " ") {
+            whitespaceStr += str[i];
+        }
+    }
+    whitespaceAmount = whitespaceStr.length;
+    return whitespaceAmount;
+}
+
+interface stringResult {
+    amountOfLowerChars: number | null;
```
Comment (viraxslot) [inline src/algorithms-task3.ts:95]: I think your functions can't (and shouldn't) return null.

File: src/algorithms-task3.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    amountOfUpperChars: number | null;
+    amountOfWhitespace: number | null;
+}
+
+function resultObj(str: string) {
+    const objResult: stringResult = {
```
Comment (viraxslot) [inline src/algorithms-task3.ts:101]: You can write just return {} and that's it. You can set the return type in the function declaration. But I believe ts can handle this type by itself.

File: src/composite-types.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+function getDel(airplane: airplaneParameters) {
+    console.log("Age = " + airplane.age + ", " + "Amount of seats = " + airplane.seats);
+}
+getDel({ age: 5, seats: 2 });
+
+//ternary operator
+let myExample = false;
+myExample ? console.log("It's TRUE") : console.log("It's FALSE");
+
+let sizeOne = 2,
```
Comment (viraxslot) [inline src/composite-types.ts:58]: I'd suggest not using the comma operator. Just 2 strings, 2 let-s. Less error-prone.

**Review by viraxslot — CHANGES_REQUESTED** [review]

File: src/composite-types.ts (hunk 9 above)
Comment (AlexandraPopovaz) [reply src/composite-types.ts]: Fixed

File: src/algorithms-task3.ts (hunk 8 above)
Comment (AlexandraPopovaz) [reply src/algorithms-task3.ts:103]: Sorry, don't understand it

File: src/algorithms-task3.ts (hunk 7 above)
Comment (AlexandraPopovaz) [reply src/algorithms-task3.ts]: Fixed

File: src/algorithms-task3.ts (hunk 6 above)
Comment (AlexandraPopovaz) [reply src/algorithms-task3.ts:16]: Because the Result should be a string consists of keys=elements

File: src/algorithms-task2.ts (hunk 4 above)
Comment (AlexandraPopovaz) [reply src/algorithms-task2.ts]: Not the same, in this case I need to change type from "number" to "any" - is it an appropriate way here?
image

File: src/algorithms-task3.ts (hunk 1 above)
Comment (AlexandraPopovaz) [reply src/algorithms-task3.ts:60]: Thanks, fixed it

File: src/algorithms-task3.ts (hunk 2 above)
Comment (AlexandraPopovaz) [reply src/algorithms-task3.ts]: Done

File: src/algorithms-task3.ts (hunk 6 above)
Comment (AlexandraPopovaz) [reply src/algorithms-task3.ts]: Thanks! Fixed

**Review by viraxslot — APPROVED** [review]

**Pull request merged**